app/code/Magento/InventoryCatalog/Ui/DataProvider/Product/Form/Modifier/SourceItems.php

```diff
@@ -136,7 +149,50 @@ public function modifyMeta(array $meta)
                     ],
                 ],
             ],
+            'children' => [
+                'assign_sources_container' => [
+                    'children' => [
+                        'assign_sources_button' => [
+                            'arguments' => [
+                                'data' => [
+                                    'config' => [
+                                        'visible' => $isMangeStock,
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+                'assigned_sources' => [
+                    'arguments' => [
+                        'data' => [
+                            'config' => [
+                                'visible' => $isMangeStock,
+                            ],
+                        ],
+                    ],
+                ],
+            ],
         ];
+
         return $meta;
     }
+
+    /**
+     * Get "isManageStock" from product, and fall back to global config in case of new product.
+     *
+     * @param ProductInterface $product
+     * @return bool
+     */
+    private function isMangeStock(ProductInterface $product): bool
+    {
+        $item = $product->getExtensionAttributes() ? $product->getExtensionAttributes()->getStockItem() : null;
```

Comment:
Pls, use `\Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface` for loading configuration
You can use default stock id (`\Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface`)

Reply:
Done.